tests/integration/test_global.py

```diff
@@ -863,6 +863,33 @@ def test_list(pixi: Path, tmp_path: Path, dummy_channel_1: str) -> None:
     )
 
 
+def test_list_with_filter(pixi: Path, tmp_path: Path, dummy_channel_1: str) -> None:
+    env = {"PIXI_HOME": str(tmp_path)}
+    manifests = tmp_path.joinpath("manifests")
+    manifests.mkdir()
+
+    # Install dummy-b from dummy-channel-1
+    verify_cli_command(
+        [
+            pixi,
+            "global",
+            "install",
+            "--channel",
+            dummy_channel_1,
+            "dummy-b==0.1.0",
+            "dummy-a==0.1.0",
+        ],
+        env=env,
+    )
+
+    # Verify list with dummy-b
+    verify_cli_command(
+        [pixi, "global", "list", "dummy-a"],
+        env=env,
+        stdout_contains=["dummy-a: 0.1.0", "dummy-a", "dummy-aa"],
+    )
```

Comment:
Could you add a test for `pixi global list --environment dummy-a dummy` and make sure that it only contains `dummy-a` and not `dummy-b`?

Reply:
yeah make sense - let me do this